### DISCUSSION TITLE
[QUESTION]

### DISCUSSION THREAD

**Discussion by arunbtech24** [longhorn/longhorn#5285] [General]
## Question
> Suggest to use https://github.com/longhorn/longhorn/discussions to ask questions.

## Environment

 - Longhorn version: 1.4.0
 - Kubernetes version: 1.26.0
 - Node config
   - OS type and version Ubuntu 22.04
   - CPU per node: 4
   - Memory per node: 16GB
   - Disk type HDD
   - Network bandwidth and latency between the nodes:
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): VMWARE

## Additional context

longhorn-manager is in crashloopbackoff during longhorn installation. Getting the below error.

fas@kuber-master:~$ kubectl logs longhorn-manager-x8kv9 -n longhorn-system
Defaulted container "longhorn-manager" out of: longhorn-manager, wait-longhorn-admission-webhook (init)
W0112 09:40:55.523134       1 client_config.go:617] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
time="2023-01-12T09:40:55Z" level=info msg="cannot list the content of the src directory /var/lib/rancher/longhorn/engine-binaries for the copy, will do nothing: failed to execute: nsenter [--mount=/host/proc/1/ns/mnt --net=/host/proc/1/ns/net bash -c ls /var/lib/rancher/longhorn/engine-binaries/*], output , stderr ls: cannot access '/var/lib/rancher/longhorn/engine-binaries/*': No such file or directory\n: exit status 2"
I0112 09:40:55.556302       1 leaderelection.go:248] attempting to acquire leader lease longhorn-system/longhorn-manager-upgrade-lock...
I0112 09:40:55.565579       1 leaderelection.go:258] successfully acquired lease longhorn-system/longhorn-manager-upgrade-lock
time="2023-01-12T09:40:55Z" level=info msg="Start upgrading"
time="2023-01-12T09:40:55Z" level=info msg="setting default-engine-image not found"
time="2023-01-12T09:40:55Z" level=error msg="Upgrade failed: upgrade API version failed: cannot create CRDAPIVersionSetting: Internal error occurred: failed calling webhook \"validator.longhorn.io\": failed to call webhook: Post \"https://longhorn-admission-webhook.longhorn-system.svc:9443/v1/webhook/validaton?timeout=10s\": Forbidden"
time="2023-01-12T09:40:55Z" level=info msg="Upgrade leader lost: kuber-worker-0"
time="2023-01-12T09:40:55Z" level=fatal msg="Error starting manager: upgrade API version failed: cannot create CRDAPIVersionSetting: Internal error occurred: failed calling webhook \"validator.longhorn.io\": failed to call webhook: Post \"https://longhorn-admission-webhook.longhorn-system.svc:9443/v1/webhook/validaton?timeout=10s\": Forbidden"
fas@kuber-master:~$

**Comment by derekbit** [comment]
Forbidden error is mostly related to your network or host's misconfiguration. Can you check these such as dns server?

**Comment by arunbtech24** [comment]
> Forbidden error is mostly related to your network or host's misconfiguration. Can you check these such as dns server?

no dns issue found.
when describe longhorn-manger pod getting the below error.

Events:
  Type     Reason       Age               From               Message
  ----     ------       ----              ----               -------
  Normal   Scheduled    39s               default-scheduler  Successfully assigned longhorn-system/longhorn-manager-qnnb9 to kuber-worker-1
  Warning  FailedMount  37s               kubelet            MountVolume.SetUp failed for volume "longhorn-grpc-tls" : failed to sync secret cache: timed out waiting for the condition
  Warning  FailedMount  37s               kubelet            MountVolume.SetUp failed for volume "kube-api-access-d57g8" : failed to sync configmap cache: timed out waiting for the condition
  Normal   Pulled       36s               kubelet            Container image "longhornio/longhorn-manager:v1.4.0" already present on machine
  Normal   Created      35s               kubelet            Created container wait-longhorn-admission-webhook
  Normal   Started      35s               kubelet            Started container wait-longhorn-admission-webhook
  Normal   Pulled       9s (x3 over 25s)  kubelet            Container image "longhornio/longhorn-manager:v1.4.0" already present on machine
  Normal   Created      9s (x3 over 24s)  kubelet            Created container longhorn-manager
  Normal   Started      8s (x3 over 24s)  kubelet            Started container longhorn-manager
  Warning  Unhealthy    8s (x3 over 24s)  kubelet            Readiness probe failed: dial tcp 10.244.2.80:9500: connect: connection refused
  Warning  BackOff      6s (x4 over 21s)  kubelet            Back-off restarting failed container longhorn-manager in pod longhorn-manager-qnnb9_longhorn-system(1ec6c21e-9119-41ed-937e-5d98ab792aea)

**Comment by derekbit** [comment]
You can check this thread https://github.com/longhorn/longhorn/issues/4462.

**Reply by arunbtech24** [reply]
Hi Derek,

Thank you so much for your response. 
Resolv.conf file on nodes.

root@kuber-master:~# cat /etc/resolv.conf
# This is /run/systemd/resolve/stub-resolv.conf managed by man:systemd-resolved(8).

# operation for /etc/resolv.conf.

nameserver 127.0.0.53
options edns0 trust-ad
search ttl-lab.local

dnsDomain: cluster.local is already set in ConfigMap 
root@kuber-master:~# kubectl edit cm kubeadm-config -n kube-system

apiVersion: v1
data:
  ClusterConfiguration: |
    apiServer:
      extraArgs:
        authorization-mode: Node,RBAC
      timeoutForControlPlane: 4m0s
    apiVersion: kubeadm.k8s.io/v1beta3
    certificatesDir: /etc/kubernetes/pki
    clusterName: kubernetes
    controllerManager: {}
    dns: {}
    etcd:
      local:
        dataDir: /var/lib/etcd
    imageRepository: registry.k8s.io
    kind: ClusterConfiguration
    kubernetesVersion: v1.26.0
    networking:
      dnsDomain: cluster.local
      podSubnet: 10.244.0.0/16
      serviceSubnet: 10.96.0.0/12
    scheduler: {}
kind: ConfigMap
metadata:
  creationTimestamp: "2023-01-02T06:53:45Z"
  name: kubeadm-config
  namespace: kube-system
  resourceVersion: "195"
  uid: 24e4b04d-9950-4f9c-a43a-bb5c6c73ba97

Below is the proxy settings on all nodes:

root@kuber-master:~# cat /etc/environment
PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin"
http_proxy=http://10.33.46.5:8080
https_proxy=http://10.33.46.5:8080
ftp_proxy=http://10.33.46.5:8080
no_proxy=10.33.57.193,.svc,.svc.cluster.local,linux-repo-20,linux-repo-20.ttl.lab,10.33.57.191,linux-pack,linux-pack.ttl.lab,127.0.0.1,localhost,10.244.0.0/16,10.33.57.75,10.33.57.76,10.33.57.77,10.96.0.0/12,10.33.57.0/24,10.0.0.0/8,0.0.0.0,10.0.0.0/8
root@kuber-master:~#

root@kuber-master:~# cat /etc/systemd/system/containerd.service.d/http-proxy.conf
[Service]
Environment="HTTP_PROXY=http://10.33.46.5:8080"
Environment="HTTPS_PROXY=http://10.33.46.5:8080"
Environment="NO_PROXY=localhost,.svc,.svc.local.cluster,127.0.0.1,10.244.0.0/16,10.33.57.75,10.33.57.76,10.33.57.77,10.96.0.0/12,10.33.57.0/24,10.33.57.0/24,0.0.0.0,10.0.0.0/8"
root@kuber-master:~#

Getting below error.
Defaulted container "longhorn-manager" out of: longhorn-manager, wait-longhorn-admission-webhook (init)
W0113 09:11:03.862471       1 client_config.go:617] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
time="2023-01-13T09:11:03Z" level=info msg="cannot list the content of the src directory /var/lib/rancher/longhorn/engine-binaries for the copy, will do nothing: failed to execute: nsenter [--mount=/host/proc/1/ns/mnt --net=/host/proc/1/ns/net bash -c ls /var/lib/rancher/longhorn/engine-binaries/*], output , stderr ls: cannot access '/var/lib/rancher/longhorn/engine-binaries/*': No such file or directory\n: exit status 2"
I0113 09:11:03.893942       1 leaderelection.go:248] attempting to acquire leader lease longhorn-system/longhorn-manager-upgrade-lock...
I0113 09:11:03.902058       1 leaderelection.go:258] successfully acquired lease longhorn-system/longhorn-manager-upgrade-lock
time="2023-01-13T09:11:03Z" level=info msg="Start upgrading"
time="2023-01-13T09:11:03Z" level=info msg="setting default-engine-image not found"
time="2023-01-13T09:11:03Z" level=error msg="Upgrade failed: upgrade API version failed: cannot create CRDAPIVersionSetting: Internal error occurred: failed calling webhook \"validator.longhorn.io\": failed to call webhook: Post \"https://longhorn-admission-webhook.longhorn-system.svc:9443/v1/webhook/validaton?timeout=10s\": Forbidden"
time="2023-01-13T09:11:03Z" level=info msg="Upgrade leader lost: kuber-worker-0"
time="2023-01-13T09:11:03Z" level=fatal msg="Error starting manager: upgrade API version failed: cannot create CRDAPIVersionSetting: Internal error occurred: failed calling webhook \"validator.longhorn.io\": failed to call webhook: Post \"https://longhorn-admission-webhook.longhorn-system.svc:9443/v1/webhook/validaton?timeout=10s\": Forbidden"

Could you please help me to resolve the issue. I am not able to figure it out the issue.

**Reply by arunbtech24** [reply]
And also installed dnsutils 

root@kuber-master:~# kubectl exec -i -t dnsutils -- nslookup longhorn-admission-webhook.longhorn-system
Server:         10.96.0.10
Address:        10.96.0.10#53

Name:   longhorn-admission-webhook.longhorn-system.svc.cluster.local
Address: 10.107.102.185

root@kuber-master:~# kubectl exec -i -t dnsutils -- nslookup longhorn-backend .longhorn-system
nslookup: couldn't get address for '.longhorn-system': not found
command terminated with exit code 10
root@kuber-master:~# kubectl exec -i -t dnsutils -- nslookup longhorn-backend.longhorn-system
Server:         10.96.0.10
Address:        10.96.0.10#53

Name:   longhorn-backend.longhorn-system.svc.cluster.local
Address: 10.109.102.108

root@kuber-master:~# kubectl exec -i -t dnsutils -- nslookup longhorn-conversion-webhook.longhorn-system
Server:         10.96.0.10
Address:        10.96.0.10#53

Name:   longhorn-conversion-webhook.longhorn-system.svc.cluster.local
Address: 10.100.83.179

root@kuber-master:~# kubectl exec -i -t dnsutils -- nslookup longhorn-engine-manager.longhorn-system
Server:         10.96.0.10
Address:        10.96.0.10#53

** server can't find longhorn-engine-manager.longhorn-system: NXDOMAIN

command terminated with exit code 1
root@kuber-master:~# kubectl exec -i -t dnsutils -- nslookup longhorn-frontend.longhorn-system
Server:         10.96.0.10
Address:        10.96.0.10#53

Name:   longhorn-frontend.longhorn-system.svc.cluster.local
Address: 10.100.176.171

root@kuber-master:~# kubectl exec -i -t dnsutils -- nslookup longhorn-recovery-backend.longhorn-system
Server:         10.96.0.10
Address:        10.96.0.10#53

Name:   longhorn-recovery-backend.longhorn-system.svc.cluster.local
Address: 10.103.245.176

root@kuber-master:~# kubectl exec -i -t dnsutils -- nslookup longhorn-replica-manager.longhorn-system
Server:         10.96.0.10
Address:        10.96.0.10#53

** server can't find longhorn-replica-manager.longhorn-system: NXDOMAIN

command terminated with exit code 1
root@kuber-master:~#

**Reply by arunbtech24** [reply]
This was a dns issue.  Reinstallation is successful after adjusting the /etc/environment file.
